rules/Php81/Rector/ClassMethod/NewInInitializerRector.php

```diff
@@ -122,9 +128,76 @@ public function provideMinPhpVersion(): int
         return PhpVersionFeature::NEW_INITIALIZERS;
     }
 
-    private function isNotNewOrWithDynamicClass(Expr $expr): bool
+    private function isFullyQualifiedClassOfNew(Expr $expr): bool
+    {
+        if (! $expr instanceof New_) {
+            return false;
+        }
+
+        return $expr->class instanceof FullyQualified;
+    }
+
+    private function shouldSkip(Expr $expr): bool
+    {
+        if (! $this->isFullyQualifiedClassOfNew($expr)) {
+            return true;
+        }
+
+        /** @var New_ $expr */
+        $args = $expr->getArgs();
+
+        foreach ($args as $arg) {
+            $value = $arg->value;
+
+            if ($this->isAllowedNew($value)) {
+                continue;
+            }
+
+            if ($value instanceof Array_ && $this->isAllowedArray($value)) {
+                continue;
+            }
+
+            if ($value instanceof Scalar) {
+                continue;
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    private function isAllowedArray(Array_ $array): bool
```

Comment:
This service is getting quite complex. Could you extra 1-2 service that detect if the expression is initializable?

Reply:
I will try.

Reply:
implemented, I created new service `ComplexNewAnalyzer` for it https://github.com/rectorphp/rector-src/pull/1737/commits/626c717437b3a72cf4346912aba3293ca4435827